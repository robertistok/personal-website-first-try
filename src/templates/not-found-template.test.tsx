import React from "react";
import { useStaticQuery } from "gatsby";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import NotFoundTemplate from "./not-found-template";
import siteMetadata from "../../jest/__fixtures__/site-metadata";

describe("<NotFoundTemplate />", () => {
  beforeEach(() => {
    (useStaticQuery as jest.Mock).mockReturnValue(siteMetadata);
  });

  it("renders", () => {
    shallow(<NotFoundTemplate />);
  });

  it("renders and matches snapshot", () => {
    const wrapper = shallow(<NotFoundTemplate />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
