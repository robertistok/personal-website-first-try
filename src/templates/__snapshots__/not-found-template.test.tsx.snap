// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundTemplate /> renders and matches snapshot 1`] = `
<Layout
  description="Test subtitle"
  title="Not Found - Test title"
>
  <Sidebar />
  <Page
    title="NOT FOUND"
  >
    <p>
      You just hit a route that doesn't exist... the sadness.
    </p>
  </Page>
</Layout>
`;
